_posts/2024-08-23-gsoc24-final-report.md

```diff
@@ -0,0 +1,133 @@
+---
+layout: post
+title: GSoC24 Final Report 
+date: 2024-08-23 22:24:25
+author: Aquila Macedo
+author_website: https://aquilamacedo.github.io/
+tags: [kw, gsoc, integration_testing]
+---
+
+Well, after spending the last few months studying and contributing to
+[kworkflow](https://kworkflow.org/) as part of **Google Summer of Code 2024**
+under the **Linux Foundation**, it’s time to catalog all the contributions made
+during this period. I can confidently say that this experience has been
+extremely enriching and has significantly advanced my development skills.
+
+# Proposal
+
+My GSoC24 proposal focused on supporting and creating new integration tests for
+**kworkflow**, which previously had only unit tests. Throughout the program, I
+worked on introducing, enhancing, and solidifying these integration tests to
+ensure more effective validation of the project's features.
+
+# Challenges and Solutions
+
+During my journey in the Google Summer of Code, I faced several challenges
+while implementing and improving integration tests for kworkflow. Initially,
+adapting the tests to run in Podman containers was a considerable challenge. I
+resolved this by thoroughly studying the [Podman
+documentation](https://docs.podman.io/en/latest/) and adjusting the test
+scripts to ensure compatibility and performance within the testing framework.
+
+Another major challenge was integrating tests for the `kw build` feature due to
+the extended time required for kernel compilation. To mitigate this issue, the
+tests were adapted to run on only one random distribution, saving both time and
+resources. I utilized Podman containers and the
+[shUnit2](https://github.com/kward/shunit2) framework to organize the tests,
+along with specific scripts to monitor and validate CPU usage with the
+`--cpu-scaling` option. This allowed `kw build` to be tested without the need
+to compile the entire kernel
+
+The integration tests for the `kw ssh` feature were more complex due to the use
+of nested containers. Adapting the tests for this scenario was challenging, but
+it also provided a valuable learning opportunity. Additionally, I had to manage
+the execution of commands with special characters inside the containers. To
+address this, I developed functions to ensure that these commands were
+correctly interpreted by the shell.
+
+I faced several challenges during my Google Summer of Code journey, but they
+were overcome with the support of mentors who were always available to answer
+my questions.
```

Comment:
Maybe sneak in this paragraph the idea that we have weekly meetings to discuss everything related to the kw community besides meetings dedicated to your specific project and that this was a great source of insights from the community.